projects/rero/ng-core/src/lib/text-read-more/text-read-more.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'ng-core-text-read-more',
+  templateUrl: './text-read-more.component.html'
+})
+export class TextReadMoreComponent implements OnInit {
+
+  /** Initial text  */
+  @Input() text: string;
+  /** Number of parts to display (depending of the unit)  */
+  @Input() limit = 50;
+  /** "Show more" label to display */
+  @Input() showMoreLabel = 'Show more';
+  /** "Show less" label to display  */
+  @Input() showLessLabel = 'Show less';
+  /** Splitting unit [ word | character] */
+  @Input() unit = 'word';
+
+  /** The text that will be display */
+  textToDisplay: string;
+  /** boolean to know if initial text is expanded  */
+  isExpanded = false;
+  /** boolean to know if limit is reached */
+  isLimitReached = true;
+  /** The truncated text  */
+  private _sTruncateText: string;
+
+  ngOnInit() {
+    this.textToDisplay = this._sTruncateText = this._truncateText();
+    if (this.text === this.textToDisplay) {
+      this.isLimitReached = false;
+    }
+  }
+
+  /**
+   * Allow to toggle content of the field
+   * @param event : Click event
+   */
+  toggleText(event: Event) {
+    event.preventDefault();
+    this.textToDisplay = (this.isExpanded)
+      ? this._sTruncateText
+      : this.text;
+    this.isExpanded = !this.isExpanded;
+  }
+
+
+  /**
+   * Truncate initial text using the defined limit and unit.
+   */
+  private _truncateText(): string {
```

Comment:
What is the behavior with a text containing html tags? I think it can causes unwanted truncation with unit "character".

Reply:
yes, in this case it causes unwanted truncation for coded HTML strings.
It's supposed to be used with "safe strings".